include/eve/module/core/regular/impl/simd/riscv/simd_cast.hpp

```diff
@@ -0,0 +1,183 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE wide<T, typename N::combined_type>
+                riscv_lmul_ext(wide<T, N> a) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, typename N::combined_type>>();
+  static_assert(N::combined_type::value != N::value);
```

Comment:
Combined type should always be twice as big, I'm not sure what's that about.
Am I missing smth?

Reply:
> Am I missing smth?

Nop. I've added assert for to be sure. Removed